src/middleware/auth/logto-helper.ts

```diff
@@ -287,58 +258,104 @@ export class LogToHelper extends OAuthProvider implements IOAuthProvider {
 		}
 	}
 
-	private async getRoleInfo(roleId: string): Promise<ICommonErrorResponse> {
-		const uri = new URL(`/api/roles/${roleId}`, process.env.LOGTO_ENDPOINT);
-		try {
-			return await this.getToLogto(uri, 'GET');
-		} catch (err) {
-			return this.returnError(StatusCodes.BAD_GATEWAY, `getRoleInfo ${err}`);
-		}
-	}
-
-	private async assignDefaultRoleForUser(userId: string, roleId: string): Promise<ICommonErrorResponse> {
-		const userInfo = await this.getUserInfo(userId);
-		const uri = new URL(`/api/users/${userId}/roles`, process.env.LOGTO_ENDPOINT);
+	async assignCustomerPlanRoles(userId: string, planType: SupportedPlanTypes): Promise<ICommonErrorResponse> {
+		const [userInfo, userRoles] = await Promise.all([this.getUserInfo(userId), this.getRolesForUser(userId)]);
+		console.log('assignCustomerPlanRoles: userInfo: ', userInfo);
+		console.log('assignCustomerPlanRoles: userRoles: ', userRoles);
 
 		if (userInfo.status !== StatusCodes.OK) {
-			return this.returnError(
-				StatusCodes.BAD_GATEWAY,
-				`Could not fetch the info about role with roleId ${roleId}`
-			);
+			return this.returnError(StatusCodes.BAD_GATEWAY, 'Could not fetch the info about the user');
 		}
 		// Means that user exists
-		if (userInfo.data.isSuspended === 'true') {
+		if (userInfo.data.isSuspended) {
 			return this.returnError(StatusCodes.FORBIDDEN, 'User is suspended');
 		}
-		// Means it's not suspended
-		const role = await this.getRoleInfo(roleId);
-		if (role.status !== StatusCodes.OK) {
-			return this.returnError(
-				StatusCodes.BAD_GATEWAY,
-				`Could not fetch the info about user with userId ${userId} because of error from authority server: ${role.error}`
-			);
+
+		if (userRoles.status !== StatusCodes.OK) {
+			return this.returnError(StatusCodes.BAD_GATEWAY, 'Could not fetch the info about user roles');
+		}
+
+		const assignedRoleIds = (userRoles.data as { id: string; name: string }[]).map((role) => role.id);
+
+		// INFO: Context
+		// We currently have two plans, Build and Test.
+		// All of our users get a "Portal" role which let's them work with our Studio Portal (UI)
+		// Test plan lets you operate with our Testnet so we assign the "Testnet" role
+		// Build plan lets you work with our Testnet and Mainnet, so we assign "Testnet" and "Mainnet" roles
+		// "build" is the superset of all the roles (testnet + mainnet)
+		const buildPlanRoleIds = [env.LOGTO_MAINNET_ROLE_ID.trim(), env.LOGTO_TESTNET_ROLE_ID.trim()];
+		const testPlanRoleId = env.LOGTO_TESTNET_ROLE_ID.trim();
+
+		const planRoleIds: string[] = [];
+		console.log('planType: ', planType);
+		if (planType === 'build') {
```

Comment:
Let's use a constant literal object definition here instead.

Reply:
I've converted this to a `String Enum` type and then compare using enum variants